### PR TITLE
groups: Add Jeremy Rickard as a SIG Release Technical Lead

### PR DESCRIPTION
**Pull request by hasheddan**
Adds @jeremyrickard to appropriate groups and OWNERS as part of sig technical lead onboarding.

(Note: Jeremy was already in the general `leads` group so he is not added in this PR)

xref: https://github.com/kubernetes/sig-release/issues/1421

/hold
/assign @justaugustus @saschagrunert

### CODE REVIEW

File: OWNERS_ALIASES [hunk 1]
```diff
@@ -222,6 +223,7 @@ aliases:
     - feiskyer # Release Manager
     - hasheddan # subproject owner / Release Manager / SIG Technical Lead
     - idealhack # Release Manager
+    - jeremyrickard # subproject owner / Release Manager / SIG Technical Lead
```
Comment (justaugustus) [inline OWNERS_ALIASES:226]:
```suggestion
    - jeremyrickard # SIG Technical Lead
```

File: OWNERS_ALIASES [hunk 2]
```diff
@@ -233,6 +235,7 @@ aliases:
...
```
Comment (justaugustus) [inline OWNERS_ALIASES:238]:
```suggestion
    - jeremyrickard # SIG Technical Lead
```

File: groups/sig-release/groups.yaml [hunk 3]
```diff
@@ -62,6 +63,7 @@ groups:
       - feiskyer@gmail.com
       - georgedanielmangum@gmail.com
       - idealhack@gmail.com
+      - jeremy.r.rickard@gmail.com
```
Comment (justaugustus) [inline groups/sig-release/groups.yaml:66]: Drop.

File: groups/sig-release/groups.yaml [hunk 4]
```diff
@@ -45,6 +45,7 @@ groups:
       ReconcileMembers: "true"
     members:
       - georgedanielmangum@gmail.com
+      - jeremy.r.rickard@gmail.com
```
Comment (justaugustus) [inline groups/sig-release/groups.yaml:48]: Drop.

**Review by justaugustus — CHANGES_REQUESTED** [review]
@hasheddan -- A few nits + let's add @jeremyrickard as an owner for release-team@kubernetes.io as well.

There are a few of us in @kubernetes/sig-release-leads that hold multiple roles...
Yourself, @saschagrunert, and I are:
- SIG leadership
- Release Managers
- RelEng subproject owners

We need to codify this somewhere, but I've been roughly going by these rules:
- All SIG Release leadership should have (owner-level) access to moderate _contact-only_ Google Groups (roughly everything that isn't prepended with `k8s-`)
- Only Release Managers and RelEng subproject owners should be on groups that grant IAM to GCP
- Only RelEng subproject owners should be on `k8s-infra-release-admins`

/hold
/retitle groups: Add Jeremy Rickard as a SIG Release Technical Lead

**Comment by hasheddan** [comment]
@justaugustus updates made :+1:

**Comment by justaugustus** [comment]
Awesome, thanks @hasheddan!
/lgtm
/approve
/hold cancel

**Pull request merged**